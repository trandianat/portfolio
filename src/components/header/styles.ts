import { css } from '@emotion/react';
import { Color } from 'utils/constants';

export const header = css({
  display: 'flex',
  flexDirection: 'column',
  gap: 36,
});

export const divider = { backgroundColor: Color.BLACK, height: 2 };

export const name = css({
  display: 'flex',
  justifyContent: 'space-between',
  '@media (max-width: 700px)': { flexDirection: 'column-reverse', gap: 24 },
  '@media (max-width: 360px)': { h1: { lineHeight: '72px', marginBottom: 12 } },
});

export const logo = css({
  backgroundColor: Color.BLACK,
  borderRadius: '50%',
  color: Color.WHITE,
  display: 'flex',
  fontSize: '2rem',
  height: 100,
  letterSpacing: 3,
  minHeight: 100,
  minWidth: 100,
  width: 100,
  '.content': {
    display: 'flex',
    flexDirection: 'column',
    gap: 6,
    margin: 'auto',
  },
  '.dot': {
    backgroundColor: Color.WHITE,
    borderRadius: '50%',
    height: 4,
    margin: '0 auto',
    width: 4,
  },
  '.line': {
    backgroundColor: Color.WHITE,
    height: 1,
    margin: '0 auto',
    width: 24,
  },
  '@media (max-width: 700px)': { margin: '0 auto' },
});

export const links = {
  '@media (min-width: 951px)': { display: 'none' },
};
