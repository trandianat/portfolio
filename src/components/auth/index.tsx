import { useAuthenticator, withAuthenticator } from '@aws-amplify/ui-react';
import { Fragment, ReactNode } from 'react';
import { Link, useLocation } from 'react-router-dom';
import * as styles from 'components/auth/styles';
import isProjects from 'utils/projects';

const Auth = ({ children }: { children: ReactNode }): JSX.Element => {
  const { signOut } = useAuthenticator();
  const { pathname } = useLocation();
  return (
    <Fragment>
      <div css={styles.navigation}>
        {isProjects(pathname) ? <div /> : <Link to="projects">â‰ª Back</Link>}
        <button onClick={signOut}>Sign out</button>
      </div>
      {children}
    </Fragment>
  );
};

export default withAuthenticator(Auth, {
  components: { SignIn: { Footer: () => <Fragment /> } },
  formFields: {
    signIn: {
      username: { label: '', placeholder: 'Username' },
      password: { label: '', placeholder: 'Password' },
    },
  },
  hideSignUp: true,
  // services: {
  //   handleSignIn: async ({
  //     username,
  //     password,
  //   }: {
  //     username: string;
  //     password: string;
  //   }) => console.log('Credentials: ' + username + ' / ' + password),
  // },
});
