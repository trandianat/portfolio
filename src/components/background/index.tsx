import { PropsWithChildren } from 'react';
import * as styles from 'components/background/styles';
import { Color } from 'utils/types';

export const Background = ({
  children,
  color = Color.WHITE,
}: PropsWithChildren<{ color?: Color }>): JSX.Element => {
  return (
    <div css={styles.background(color)}>
      <div css={styles.container}>{children}</div>
    </div>
  );
};
