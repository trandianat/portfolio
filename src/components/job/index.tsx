import * as styles from 'components/job/styles';

export const Job = ({
  role,
  company,
  start,
  end,
  skills,
  descriptions,
  indent = false,
}: {
  role: string;
  company: string;
  start: string;
  end: string;
  skills: string[];
  descriptions: string[];
  indent?: boolean;
}): JSX.Element => (
  <div css={styles.job(indent)}>
    <div css={styles.dates}>
      <div css={styles.range}>
        <div className="sans">{end}</div>
        <div className="sans">{start}</div>
      </div>
      <div css={styles.timeline}>
        <div css={styles.circle} />
        <div css={styles.line} />
        <div css={styles.circle} />
      </div>
    </div>
    <div css={styles.role}>
      <h2>{role}</h2>
      <p className="sans">{company}</p>
      <div css={styles.skills}>
        {skills.map((skill: string) => (
          <div css={styles.skill} key={skill}>
            {skill}
          </div>
        ))}
      </div>
      <ul>
        {descriptions.map((description: string) => (
          <li key={description}>{description}</li>
        ))}
      </ul>
    </div>
  </div>
);
