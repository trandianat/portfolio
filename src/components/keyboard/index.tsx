import { useEffect, useState } from 'react';
import * as styles from 'components/keyboard/styles';
import { Keyboard as Key } from 'utils/constants';

export const Keyboard = ({
  disabled,
  guesses,
  handleSubmit,
  input,
  setInput,
}: {
  disabled: boolean;
  guesses: string[];
  handleSubmit: any;
  input: string;
  setInput: any;
}): JSX.Element => {
  const [rows, setRows] = useState<Array<string[]>>([]);

  const isAction = (key: string): boolean =>
    [Key.DELETE, Key.ENTER].includes(key as Key);
  const isDelete = (key: string): boolean => key === Key.DELETE;
  const isEnter = (key: string): boolean => key === Key.ENTER;

  const setLetterClass = (key: string, rowIndex: number): string => {
    let className = 'letter';
    if (isAction(key)) {
      className += ' action';
      if (rowIndex < rows.length - 1) {
        className += ' primary';
      }
    } else {
      const used = guesses.map(guess => [...guess]).flat();
      if (used.includes(key)) {
        className += ' used';
      }
    }
    return className;
  };

  useEffect(() => {
    const row1 = ['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'];
    const row2 = ['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'];
    const row3 = [Key.ENTER, 'Z', 'X', 'C', 'V', 'B', 'N', 'M', Key.DELETE];
    const row4 = [Key.ENTER, Key.DELETE];
    setRows([row1, row2, row3, row4]);
  }, []);

  return (
    <div css={styles.keyboard}>
      {rows.map((row, index) => (
        <div className={`row row-${index + 1}`} key={`row-${index + 1}`}>
          {row.map((key: string) => (
            <button
              className={setLetterClass(key, index)}
              key={key}
              onClick={() => {
                if (isDelete(key)) {
                  setInput((input: string) => input.slice(0, -1));
                } else if (isEnter(key)) {
                  handleSubmit();
                } else {
                  setInput((input: string) => input + key);
                }
              }}
              {...(((isDelete(key) && !input.length) ||
                (isEnter(key) && input.length < 5) ||
                (!isAction(key) && input.length === 5) ||
                disabled) && {
                disabled: true,
              })}
            >
              {key}
            </button>
          ))}
        </div>
      ))}
    </div>
  );
};
