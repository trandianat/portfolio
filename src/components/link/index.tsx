import { PropsWithChildren, ReactNode } from 'react';
import * as styles from 'components/link/styles';
import { Position, Size } from 'utils/constants';

export const Link = ({
  children,
  icon,
  iconGap = Size.SMALL,
  iconPosition = Position.RIGHT,
  text,
  url,
  ...rest
}: PropsWithChildren<{
  icon?: ReactNode;
  iconGap?: Size;
  iconPosition?: Position;
  text: string;
  url: string;
  [rest: string]: unknown;
}>): JSX.Element => (
  <a
    css={styles.link(iconGap, iconPosition)}
    href={url}
    target="_blank"
    {...rest}
  >
    {iconPosition === Position.LEFT && <span className="icon">{icon}</span>}
    <span className="text">{text}</span>
    {iconPosition === Position.RIGHT && <span className="icon">{icon}</span>}
  </a>
);
