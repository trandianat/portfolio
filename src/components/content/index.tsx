import { Fragment, ReactNode } from 'react';
import { Background } from 'components/background';
import { Header } from 'components/header';
import { Navigation } from 'components/navigation';
import { Color } from 'utils/constants';

export const Content = ({ children }: { children: ReactNode }): JSX.Element => (
  <Fragment>
    <Header />
    <Background color={Color.GRAY}>
      <Navigation />
      {children}
    </Background>
  </Fragment>
);
