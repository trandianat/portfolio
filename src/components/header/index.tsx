import { ReactNode } from 'react';
import { Background } from 'components/background';
import * as styles from 'components/header/styles';

export const Header = ({ children }: { children: ReactNode }): JSX.Element => (
  <Background>
    <div css={styles.header}>
      <div css={styles.divider} />
      <div css={styles.name}>
        <div>
          <h1>Diana Tran</h1>
          <h2 className="sans">Software Engineer & Manager</h2>
        </div>
        <div css={styles.logo}>
          <div className="content">
            <div className="dot" />
            <div>DT</div>
            <div className="line" />
          </div>
        </div>
      </div>
      {children}
    </div>
  </Background>
);
