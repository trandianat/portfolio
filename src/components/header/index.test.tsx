import { ReactNode } from 'react';
import { create } from 'react-test-renderer';
import { Header } from 'components/header';

jest.mock('components/background', () => ({
  Background: ({ children }: { children: ReactNode }) => (
    <div className="background">{children}</div>
  ),
}));

describe('Header', () => {
  it('should match the snapshot', () => {
    expect(create(<Header />)).toMatchSnapshot();
  });
});
