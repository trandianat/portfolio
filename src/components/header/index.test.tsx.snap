// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 36px;
}

.emotion-1 {
  background-color: var(--black);
  height: 2px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 700px) {
  .emotion-2 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    gap: 24px;
  }
}

@media (max-width: 360px) {
  .emotion-2 h1 {
    line-height: 72px;
    margin-bottom: 12px;
  }
}

.emotion-3 {
  background-color: var(--black);
  border-radius: 50%;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2rem;
  height: 100px;
  letter-spacing: 3px;
  min-height: 100px;
  min-width: 100px;
  width: 100px;
}

.emotion-3 .content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 6px;
  margin: auto;
}

.emotion-3 .dot {
  background-color: var(--white);
  border-radius: 50%;
  height: 4px;
  margin: 0 auto;
  width: 4px;
}

.emotion-3 .line {
  background-color: var(--white);
  height: 1px;
  margin: 0 auto;
  width: 24px;
}

@media (max-width: 700px) {
  .emotion-3 {
    margin: 0 auto;
  }
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  gap: 4px;
  vertical-align: bottom;
}

@media (min-width: 701px) {
  .emotion-4 {
    display: none;
  }
}

<div
  className="background"
>
  <div
    className="emotion-0"
  >
    <div
      className="emotion-1"
    />
    <div
      className="emotion-2"
    >
      <div>
        <h1>
          Diana Tran
        </h1>
        <p
          className="sans"
        >
          Software Engineer
        </p>
      </div>
      <div
        className="emotion-3"
      >
        <div
          className="content"
        >
          <div
            className="dot"
          />
          <div>
            DT
          </div>
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <p>
      Results-driven software engineer with 8 years of full-stack experience and 1 year of people-management experience. Seeking a role that highlights my passion for developing seamless and confidence-boosting user experiences. Graduated from Boston College in 2014 with a bachelor's in Computer Science.
       
      <a
        className="emotion-4"
        href="https://github.com/trandianat"
        target="_blank"
      >
        Github
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3c-1.093 0-2 .907-2 2v14c0 1.093.907 2 2 2h14c1.093 0 2-.907 2-2v-7h-2v7H5V5h7V3H5zm9 0v2h3.586l-9.293 9.293 1.414 1.414L19 6.414V10h2V3h-7z"
          />
        </svg>
      </a>
    </p>
  </div>
</div>
`;
